workos/audit_trail.py

```diff
@@ -48,21 +50,126 @@ def create_event(self, event, idempotency_key=None):
+        actor_id=None,
+        target_name=None,
+        target_id=None,
+        occurred_at=None,
+        occurred_at_gt=None,
+        occurred_at_gte=None,
+        occurred_at_lt=None,
+        occurred_at_lte=None,
+        search=None,
+    ):
+        """Filter for Audit Trail Events.
+
+        Kwargs:
+            before (str) - Event ID to look before
+            after (str) - Event ID to look after
+            limit (int) - Number of Events to return
+            group (str|list) - Group or groups to filter for
+            action (str|list) - Action or actions to filter for
+            action_type (str|list) - Action type or types to filter for
+            actor_name (str|list) - Actor name or name to filter for
+            actor_id (str|list) - Actor ID or IDs to filter for
+            target_name (str|list) - Target name or names to filter for
+            target_id (str|list) - Target ID or IDs to filter for
+            occurred_at (str) - ISO-8601 datetime of when an event occurred
+            occurred_at_gt (str) - ISO-8601 datetime of when an event occurred after
+            occurred_at_gte (str) - ISO-8601 datetime of when an event occurred at or after
+            occurred_at_lt (str) - ISO-8601 datetime of when an event occurred before
+            occurred_at_lte (str) - ISO-8601 datetime of when an event occured at or before
+            search (str) - Keyword search
+
+        Returns:
+            tuple
+                list - List of WorkOSEvent objects
+                string - Event ID to use as before cursor
+                string - Event ID to use as after cursor
+        """
+        if before and after:
+            raise ValueError("Specify either before or after")
+
+        params = {
+            "before": before,
+            "after": after,
+            "limit": limit,
+        }
+
+        if group is not None:
+            params["group"] = list(group)
+
+        if action is not None:
+            params["action"] = list(action)
+
+        if action_type is not None:
+            params["action_type"] = list(action_type)
+
+        if actor_name is not None:
+            params["actor_name"] = list(actor_name)
+
+        if actor_id is not None:
+            params["actor_id"] = list(actor_id)
+
+        if target_name is not None:
+            params["target_name"] = list(target_name)
+
+        if target_id is not None:
+            params["target_id"] = list(target_id)
+
+        if occurred_at is not None:
+            params["occurred_at"] = occurred_at
+        else:
+            if occurred_at_gte is not None:
+                params["occurred_at_gte"] = occurred_at_gte
+            elif occurred_at_gt is not None:
+                params["occurred_at_gt"] = occurred_at_gt
+
+            if occurred_at_lte is not None:
+                params["occurred_at_lte"] = occurred_at_lte
+            elif occurred_at_lt is not None:
+                params["occurred_at_lt"] = occurred_at_lt
```

Comment:
is this validation handled by our api? if so, is it necessary to do so here?

Reply:
The validation isn't on the app side. I also think the database should query for the right thing either way given the extraneous args so this is moreso me wanting a cleaner set of filters passed over the wire